chainstate/src/lib.rs

```diff
@@ -54,10 +54,10 @@ impl subsystem::Subsystem for Box<dyn ChainstateInterface> {}
 
 type ChainstateHandle = subsystem::Handle<Box<dyn ChainstateInterface>>;
 
-pub fn make_chainstate(
+pub fn make_chainstate<S: chainstate_storage::BlockchainStorage + 'static>(
```

Comment:
Is there a way to make this proper lifetime instead of static?

Reply:
I think that the `static` lifetime isn't an issue here, because we don't use references anyway: chainstate owns a storage.

Reply:
Why can't we relate the lifetime to chainstate lifetime instead of static?

Reply:
The `'static` thing here declares that `S` does not have any references outside of stuff it owns, i.e. it is self-contained. In some sense, having `'static` here is maximally limiting to the storage implementation (it can't have references to non-static things) but maximally liberating to clients, that is chainstate and anything that uses it (its allowed to take references to it and move it with the least amount of restriction). I think this is a good starting point but we can adjust it if the need arises.

> Why can't we relate the lifetime to chainstate lifetime instead of static?

How do you think they should relate? It's should only be needed if there are going to be references between the two. Since chainstate owns storage, I don't see what would it help with.